src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdServerService.cs

```diff
@@ -497,11 +497,14 @@ private async Task<ImmutableArray<(string path, X509Certificate2 certificate)>>
                                 X509KeyStorageFlags.EphemeralKeySet :
                                 X509KeyStorageFlags.MachineKeySet;
 #else
-                                X509KeyStorageFlags.MachineKeySet;
+                                X509KeyStorageFlags.MachineKeySet |
```

Comment:
Assuming you're going to target .NET Core, you'll likely hit the other branch (in the #ifdef). You may want to adopt the same code path for both .NET Standard and .NET Core, assuming we can make it work on every possible environment.